README.md
# Ng2 Grid
[![Build Status](https://travis-ci.org/raketasoft/ng2-grid.svg?branch=master)](https://travis-ci.org/raketasoft/ng2-grid)

Lightweight data grid component for Angular 2.

## Features

* Sorting
* Filtering
* Paging
* Selection
* Templating
* Remote Data Access

## Installation

Start by installing npm dependencies:

```bash
> npm install raketasoft-ng2-grid --save
> npm install lodash --save
```

Load CSS:

```html
<link rel="stylesheet" type="text/css" href="node_modules/raketasoft-ng2-grid/src/assets/ng2-grid.css">
```

The only way to include the component right now is to use module loader like
[SystemJS](https://github.com/systemjs/systemjs). Example configuration:

```javascript
System.config({
    defaultJSExtensions: true,
    map: {
        'raketasoft-ng2-grid': 'node_modules/raketasoft-ng2-grid',
        'lodash': 'node_modules/lodash'
    }
});
```

Then import and include in your component's directives:

```typescript
import { GridComponent, GridColumnComponent, GridOptions } from 'raketasoft-ng2-grid';

@Component({
    directives: [GridComponent, GridColumnComponent]
})
export class MyComponent {
    ...
}
```

## Demo

Configuration examples could be found under the demo application:
[Demo](https://github.com/raketasoft/ng2-grid/tree/master/demo)

![Image of Ng2 Grid Demo](https://github.com/raketasoft/ng2-grid/blob/master/demo/assets/ng2-grid-demo.png)

## License

MIT


CHANGELOG.md
<a name="2.2.0"></a>
# 2.2.0 (2019-03-07)

### Features

* **grid:** Added callback for custom filtering
* **grid:** Added option for sticky scrollbar when grid has horizontal scroll
* **grid:** Added option for custom templates in the column headers

<a name="1.0.2"></a>
# 1.0.2 (2018-11-15)

### Bug Fixes

* **grid:** Fix issue #43 pageSize param not passed correctly


<a name="1.0.1"></a>
# 1.0.1 (2018-07-31)

### Bug Fixes

* **grid:** Fix issue with grid selection not working properly


<a name="1.0.0"></a>
# 1.0.0 (2018-05-16)

### Features

* **grid:** Bump to Angular 5.x
* **grid:** Migrate to HttpClientModule

### Bug Fixes

* **build:** Fix Chrome version in CI build


<a name="1.0.0-beta.9"></a>
# 1.0.0-beta.9 (2018-02-21)

### Features

* **build:** Bump node version for CI to latest
* **build:** Fix tslint warnings

### Bug Fixes

* **grid:** Fix issue #37
* **grid:** Fix issue with sticky header


<a name="1.0.0-beta.8"></a>
# 1.0.0-beta.8 (2017-08-29)

### Features

* **package:** Update Angular to 4.x along with related dependencies
* **build:** Bump node version for CI to 6.x
* **build:** Fix tslint rules


<a name="1.0.0-beta.7"></a>
# 1.0.0-beta.7 (2017-06-19)

### Features

* **grid:** Add single selection support for grid items
* **package:** Add package-lock.json file


<a name="1.0.0-beta.6"></a>
# 1.0.0-beta.6 (2017-05-18)

### Features

* **grid:** Add support for transforming json data to model classes (issue #2)
* **grid:** Add getter/setter for grid options (issue #29)
* **grid:** Add support for cell style callback
* **build:** Configure test build and Travis CI

### Bug Fixes

* **grid:** Fix selection issue
* **build:** Fix lint rules/errors


<a name="1.0.0-beta.5"></a>
# 1.0.0-beta.5 (2017-03-14)

### Bug Fixes

* **grid:** Fix issue #27
* **grid:** Fix issue with page button event propagation


<a name="1.0.0-beta.4"></a>
# 1.0.0-beta.4 (2017-02-22)

### Bug Fixes

* **grid:** Fix selection issue
* **grid:** Fix static data filtering issue


<a name="1.0.0-beta.3"></a>
# 1.0.0-beta.3 (2017-02-14)

### Features

* **demo:** Add node server for demo data

### Bug Fixes

* **grid:** Fix issue with change detector ref
* **grid:** Add missing jsdoc
* **test:** Fix demo component spec


<a name="1.0.0-beta.2"></a>
# 1.0.0-beta.2 (2017-02-13)

### Features

* **build:** Add TypeScript definitions to build


<a name="1.0.0-beta.1"></a>
# 1.0.0-beta.1 (2017-02-10)

### Features

* **grid:** Update grid component to work with Angular 2.4.0 and bumped related dependencies
* **test:** Add support for unit and e2e tests (actual tests not yet implemented)


<a name="1.0.0-alpha.17"></a>
# 1.0.0-alpha.17 (2017-02-03)

### Features

* **grid:** Allow filtering/sorting by columns that are not displayed in grid (issue #21). NOTE: This feature breaks backwards compatability with "type" property of grid columns. A new "filterType" property has been introduced to define how filter is displayed in UI and "type" property is now reserved only for filtering/sorting logic.

### Bug Fixes

* **grid:** Fix issue with full table width calculation


<a name="1.0.0-alpha.16"></a>
# 1.0.0-alpha.16 (2017-01-24)

### Features

* **grid:** Add support for grid events (issue #3)

### Bug Fixes

* **grid:** Fix grid resizing/rendering issues
* **grid:** Fix issue with data provider (issue #20)


<a name="1.0.0-alpha.15"></a>
# 1.0.0-alpha.15 (2017-01-12)

### Features

* **grid-column:** Add support for "number" type column, including updates to filtering, sorting and validation (issue #15)
* **grid:** Add public methods for clearing filters (issue #16)

### Bug Fixes

* **grid:** Fix bug with fixed header null offset
* **grid:** Fix various issues with scrolling/dragging the grid
* **grid:** Fix bug with select all items checkbox


<a name="1.0.0-alpha.14"></a>
# 1.0.0-alpha.14 (2016-11-30)

### Features

* **grid-column:** Add css class property to grid column


<a name="1.0.0-alpha.13"></a>
# 1.0.0-alpha.13 (2016-11-29)

### Features

* **grid:** Add support for horizontal body scroll using mouse drag

### Bug Fixes

* **grid:** Fixed issue #12


<a name="1.0.0-alpha.12"></a>
# 1.0.0-alpha.12 (2016-11-15)

### Features

* **grid:** Add API method for clearing selection
* **grid:** Add option for default filtering column
* **grid:** Add option for required filters
* **grid:** Add fixed header option
* **demo:** Update demo with new features

### Bug Fixes

* **grid:** Fixed issue #9
* **grid:** Fixed issue #10


<a name="1.0.0-alpha.11"></a>
# 1.0.0-alpha.11 (2016-09-08)

### Features

* **grid:** Add support for preserving selection during rendering (page change, filtering, etc.)
* **grid-options:** Add new configuration option (*uniqueId*) for unique ID field for data rows. Unique ID is used to preserve selection during rendering. Defaults to *id*.
* **demo:** Update demo with new selection examples


<a name="1.0.0-alpha.10"></a>
# 1.0.0-alpha.10 (2016-06-16)

### Bug Fixes

* **grid-data-provider:** Fix issue with default initialization of requestParams


<a name="1.0.0-alpha.9"></a>
# 1.0.0-alpha.9 (2016-06-16)

### Features

* **docs:** Update README

### Bug Fixes

* **grid:** Remove css autoload


<a name="1.0.0-alpha.8"></a>
# 1.0.0-alpha.8 (2016-06-15)

### Bug Fixes

* **grid:** Move template definition inline for publishing with the js file


<a name="1.0.0-alpha.7"></a>
# 1.0.0-alpha.7 (2016-06-14)

### Features

* **grid:** Move filtering logic outside data provider, update grid API

### Bug Fixes

* **grid-column:** Fix issue with select filter option values


<a name="1.0.0-alpha.6"></a>
# 1.0.0-alpha.6 (2016-06-12)

### Features

* **grid-column:** Add column type support and select/drop-down filters
* **grid-options:** Add row style customization support using callback function
* **grid-options:** Add row selection style customization
* **grid:** Add support for selecting item on row click
* **demo:** Update demo examples

### Bug Fixes


demo/data.ts
export default [
  {
    id: 1,
    name: 'Otto Clay',
    age: 61,
    country: { name: 'China', id: 3 },
    address: 'Ap #897-1459 Quam Avenue',
    isMarried: false,
    interests: [
      {
        id: 4,
        name: 'Sports'
      },
      {
        id: 3,
        name: 'Painting'
      },
      {
        id: 1,
        name: 'Dancing'
      },
    ]
  },
  {
    id: 2,
    name: 'Connor Johnston',
    age: 22,
    country: { name: 'Russia', id: 5 },
    address: 'Ap #370-4647 Dis Av.',
    isMarried: false
  },
  {
    id: 3,
    name: 'Lacey Hess',
    age: 29,
    country: { name: 'Russia', id: 5 },
    address: 'Ap #365-8835 Integer St.',
    isMarried: false
  },
  {
    id: 4,
    name: 'Timothy Henson',
    age: 78,
    country: { name: 'United States', id: 7 },
    address: '911-5143 Luctus Ave',
    isMarried: false
  },
  {
    id: 5,
    name: 'Ramona Benton',
    age: 43,
    country: { name: 'Brazil', id: 1 },
    address: 'Ap #614-689 Vehicula Street',
    isMarried: true,
    interests: [
      {
        id: 2,
        name: 'Photography'
      },
      {
        id: 3,
        name: 'Painting'
      },
      {
        id: 6,
        name: 'Fashion'
      },
    ]
  },
  {
    id: 6,
    name: 'Ezra Tillman',
    age: 51,
    country: { name: 'United States', id: 7 },
    address: 'P.O. Box 738, 7583 Quisque St.',
    isMarried: true
  },
  {
    id: 7,
    name: 'Dante Carter',
    age: 59,
    country: { name: 'United States', id: 7 },
    address: 'P.O. Box 976, 6316 Lorem, St.',
    isMarried: false
  },
  {
    id: 8,
    name: 'Christopher Mcclure',
    age: 58,
    country: { name: 'United States', id: 7 },
    address: '847-4303 Dictum Av.',
    isMarried: true
  },
  {
    id: 9,
    name: 'Ruby Rocha',
    age: 62,
    country: { name: 'Canada', id: 2 },
    address: '5212 Sagittis Ave',
    isMarried: false
  },
  {
    id: 10,
    name: 'Imelda Hardin',
    age: 39,
    country: { name: 'Brazil', id: 1 },
    address: '719-7009 Auctor Av.',
    isMarried: false
  },
  {
    id: 11,
    name: 'Jonah Johns',
    age: 28,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 939, 9310 A Ave',
    isMarried: false,
    interests: [
      {
        id: 2,
        name: 'Photography'
      },
      {
        id: 5,
        name: 'Traveling'
      },
      {
        id: 6,
        name: 'Fashion'
      },
    ]
  },
  {
    id: 12,
    name: 'Herman Rosa',
    age: 49,
    country: { name: 'Russia', id: 5 },
    address: '718-7162 Molestie Av.',
    isMarried: true
  },
  {
    id: 13,
    name: 'Arthur Gay',
    age: 20,
    country: { name: 'Russia', id: 5 },
    address: '5497 Neque Street',
    isMarried: false
  },
  {
    id: 14,
    name: 'Xena Wilkerson',
    age: 63,
    country: { name: 'United States', id: 7 },
    address: 'Ap #303-6974 Proin Street',
    isMarried: true
  },
  {
    id: 15,
    name: 'Lilah Atkins',
    age: 33,
    country: { name: 'Brazil', id: 1 },
    address: '622-8602 Gravida Ave',
    isMarried: true
  },
  {
    id: 16,
    name: 'Malik Shepard',
    age: 59,
    country: { name: 'United States', id: 7 },
    address: '967-5176 Tincidunt Av.',
    isMarried: false
  },
  {
    id: 17,
    name: 'Keely Silva',
    age: 24,
    country: { name: 'United States', id: 7 },
    address: 'P.O. Box 153, 8995 Praesent Ave',
    isMarried: false
  },
  {
    id: 18,
    name: 'Hunter Pate',
    age: 73,
    country: { name: 'Russia', id: 5 },
    address: 'P.O. Box 771, 7599 Ante, Road',
    isMarried: false
  },
  {
    id: 19,
    name: 'Mikayla Roach',
    age: 55,
    country: { name: 'Brazil', id: 1 },
    address: 'Ap #438-9886 Donec Rd.',
    isMarried: true
  },
  {
    id: 20,
    name: 'Upton Joseph',
    age: 48,
    country: { name: 'France', id: 4 },
    address: 'Ap #896-7592 Habitant St.',
    isMarried: true
  },
  {
    id: 21,
    name: 'Jeanette Pate',
    age: 59,
    country: { name: 'Canada', id: 2 },
    address: 'P.O. Box 177, 7584 Amet, St.',
    isMarried: false
  },
  {
    id: 22,
    name: 'Kaden Hernandez',
    age: 79,
    country: { name: 'United Kingdom', id: 6 },
    address: '366 Ut St.',
    isMarried: true
  },
  {
    id: 23,
    name: 'Kenyon Stevens',
    age: 20,
    country: { name: 'United Kingdom', id: 6 },
    address: 'P.O. Box 704, 4580 Gravida Rd.',
    isMarried: false
  },
  {
    id: 24,
    name: 'Jerome Harper',
    age: 31,
    country: { name: 'Brazil', id: 1 },
    address: '2464 Porttitor Road',
    isMarried: false
  },
  {
    id: 25,
    name: 'Jelani Patel',
    age: 36,
    country: { name: 'Canada', id: 2 },
    address: 'P.O. Box 541, 5805 Nec Av.',
    isMarried: true
  },
  {
    id: 26,
    name: 'Keaton Oconnor',
    age: 21,
    country: { name: 'United States', id: 7 },
    address: 'Ap #657-1093 Nec, Street',
    isMarried: false
  },
  {
    id: 27,
    name: 'Bree Johnston',
    age: 31,
    country: { name: 'Canada', id: 2 },
    address: '372-5942 Vulputate Avenue',
    isMarried: false
  },
  {
    id: 28,
    name: 'Maisie Hodges',
    age: 70,
    country: { name: 'Russia', id: 5 },
    address: 'P.O. Box 445, 3880 Odio, Rd.',
    isMarried: false
  },
  {
    id: 29,
    name: 'Kuame Calhoun',
    age: 39,
    country: { name: 'Canada', id: 2 },
    address: 'P.O. Box 609, 4105 Rutrum St.',
    isMarried: true
  },
  {
    id: 30,
    name: 'Carlos Cameron',
    age: 38,
    country: { name: 'Brazil', id: 1 },
    address: 'Ap #215-5386 A, Avenue',
    isMarried: false
  },
  {
    id: 31,
    name: 'Fulton Parsons',
    age: 25,
    country: { name: 'Russia', id: 5 },
    address: 'P.O. Box 523, 3705 Sed Rd.',
    isMarried: false
  },
  {
    id: 32,
    name: 'Wallace Christian',
    age: 43,
    country: { name: 'United Kingdom', id: 6 },
    address: '416-8816 Mauris Avenue',
    isMarried: true
  },
  {
    id: 33,
    name: 'Caryn Maldonado',
    age: 40,
    country: { name: 'United States', id: 7 },
    address: '108-282 Nonummy Ave',
    isMarried: false
  },
  {
    id: 34,
    name: 'Whilemina Frank',
    age: 20,
    country: { name: 'Russia', id: 5 },
    address: 'P.O. Box 681, 3938 Egestas. Av.',
    isMarried: true
  },
  {
    id: 35,
    name: 'Emery Moon',
    age: 41,
    country: { name: 'France', id: 4 },
    address: 'Ap #717-8556 Non Road',
    isMarried: true
  },
  {
    id: 36,
    name: 'Price Watkins',
    age: 35,
    country: { name: 'France', id: 4 },
    address: '832-7810 Nunc Rd.',
    isMarried: false
  },
  {
    id: 37,
    name: 'Lydia Castillo',
    age: 59,
    country: { name: 'Russia', id: 5 },
    address: '5280 Placerat, Ave',
    isMarried: true
  },
  {
    id: 38,
    name: 'Lawrence Conway',
    age: 53,
    country: { name: 'United States', id: 7 },
    address: 'Ap #452-2808 Imperdiet St.',
    isMarried: false
  },
  {
    id: 39,
    name: 'Kalia Nicholson',
    age: 67,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 871, 3023 Tellus Road',
    isMarried: true
  },
  {
    id: 40,
    name: 'Brielle Baxter',
    age: 45,
    country: { name: 'United Kingdom', id: 6 },
    address: 'Ap #822-9526 Ut, Road',
    isMarried: true
  },
  {
    id: 41,
    name: 'Valentine Brady',
    age: 72,
    country: { name: 'Russia', id: 5 },
    address: '8014 Enim. Road',
    isMarried: true
  },
  {
    id: 42,
    name: 'Rebecca Gardner',
    age: 57,
    country: { name: 'France', id: 4 },
    address: '8655 Arcu. Road',
    isMarried: true
  },
  {
    id: 43,
    name: 'Vladimir Tate',
    age: 26,
    country: { name: 'United States', id: 7 },
    address: '130-1291 Non, Rd.',
    isMarried: true
  },
  {
    id: 44,
    name: 'Vernon Hays',
    age: 56,
    country: { name: 'France', id: 4 },
    address: '964-5552 In Rd.',
    isMarried: true
  },
  {
    id: 45,
    name: 'Allegra Hull',
    age: 22,
    country: { name: 'France', id: 4 },
    address: '245-8891 Donec St.',
    isMarried: true
  },
  {
    id: 46,
    name: 'Hu Hendrix',
    age: 65,
    country: { name: 'Russia', id: 5 },
    address: '428-5404 Tempus Ave',
    isMarried: true
  },
  {
    id: 47,
    name: 'Kenyon Battle',
    age: 32,
    country: { name: 'Canada', id: 2 },
    address: '921-6804 Lectus St.',
    isMarried: false
  },
  {
    id: 48,
    name: 'Gloria Nielsen',
    age: 24,
    country: { name: 'France', id: 4 },
    address: 'Ap #275-4345 Lorem, Street',
    isMarried: true
  },
  {
    id: 49,
    name: 'Illiana Kidd',
    age: 59,
    country: { name: 'Canada', id: 2 },
    address: '7618 Lacus. Av.',
    isMarried: false
  },
  {
    id: 50,
    name: 'Adria Todd',
    age: 68,
    country: { name: 'China', id: 3 },
    address: '1889 Tincidunt Road',
    isMarried: false
  },
  {
    id: 51,
    name: 'Kirsten Mayo',
    age: 71,
    country: { name: 'United States', id: 7 },
    address: '100-8640 Orci, Avenue',
    isMarried: false
  },
  {
    id: 52,
    name: 'Willa Hobbs',
    age: 60,
    country: { name: 'China', id: 3 },
    address: 'P.O. Box 323, 158 Tristique St.',
    isMarried: false
  },
  {
    id: 53,
    name: 'Alexis Clements',
    age: 69,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 176, 5107 Proin Rd.',
    isMarried: false
  },
  {
    id: 54,
    name: 'Akeem Conrad',
    age: 60,
    country: { name: 'Canada', id: 2 },
    address: '282-495 Sed Ave',
    isMarried: true
  },
  {
    id: 55,
    name: 'Montana Silva',
    age: 79,
    country: { name: 'China', id: 3 },
    address: 'P.O. Box 120, 9766 Consectetuer St.',
    isMarried: false
  },
  {
    id: 56,
    name: 'Kaseem Hensley',
    age: 77,
    country: { name: 'China', id: 3 },
    address: 'Ap #510-8903 Mauris. Av.',
    isMarried: true
  },
  {
    id: 57,
    name: 'Christopher Morton',
    age: 35,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 234, 3651 Sodales Avenue',
    isMarried: false
  },
  {
    id: 58,
    name: 'Wade Fernandez',
    age: 49,
    country: { name: 'China', id: 3 },
    address: '740-5059 Dolor. Road',
    isMarried: true
  },
  {
    id: 59,
    name: 'Illiana Kirby',
    age: 31,
    country: { name: 'Canada', id: 2 },
    address: '527-3553 Mi Ave',
    isMarried: false
  },
  {
    id: 60,
    name: 'Kimberley Hurley',
    age: 65,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 637, 9915 Dictum St.',
    isMarried: false
  },
  {
    id: 61,
    name: 'Arthur Olsen',
    age: 74,
    country: { name: 'Brazil', id: 1 },
    address: '887-5080 Eget St.',
    isMarried: false
  },
  {
    id: 62,
    name: 'Brody Potts',
    age: 59,
    country: { name: 'Canada', id: 2 },
    address: 'Ap #577-7690 Sem Road',
    isMarried: false
  },
  {
    id: 63,
    name: 'Dillon Ford',
    age: 60,
    country: { name: 'United States', id: 7 },
    address: 'Ap #885-9289 A, Av.',
    isMarried: true
  },
  {
    id: 64,
    name: 'Hannah Juarez',
    age: 61,
    country: { name: 'Canada', id: 2 },
    address: '4744 Sapien, Rd.',
    isMarried: true
  },
  {
    id: 65,
    name: 'Vincent Shaffer',
    age: 25,
    country: { name: 'Canada', id: 2 },
    address: '9203 Nunc St.',
    isMarried: true
  },
  {
    id: 66,
    name: 'George Holt',
    age: 27,
    country: { name: 'China', id: 3 },
    address: '4162 Cras Rd.',
    isMarried: false
  },
  {
    id: 67,
    name: 'Tobias Bartlett',
    age: 74,
    country: { name: 'France', id: 4 },
    address: '792-6145 Mauris St.',
    isMarried: true
  },
  {
    id: 68,
    name: 'Xavier Hooper',
    age: 35,
    country: { name: 'United States', id: 7 },
    address: '879-5026 Interdum. Rd.',
    isMarried: false
  },
  {
    id: 69,
    name: 'Declan Dorsey',
    age: 31,
    country: { name: 'Canada', id: 2 },
    address: 'Ap #926-4171 Aenean Road',
    isMarried: true
  },
  {
    id: 70,
    name: 'Clementine Tran',
    age: 43,
    country: { name: 'France', id: 4 },
    address: 'P.O. Box 176, 9865 Eu Rd.',
    isMarried: true
  },
  {
    id: 71,
    name: 'Pamela Moody',
    age: 55,
    country: { name: 'China', id: 3 },
    address: '622-6233 Luctus Rd.',
    isMarried: true
  },
  {
    id: 72,
    name: 'Julie Leon',
    age: 43,
    country: { name: 'China', id: 3 },
    address: 'Ap #915-6782 Sem Av.',
    isMarried: true
  },
  {
    id: 73,
    name: 'Shana Nolan',
    age: 79,
    country: { name: 'Brazil', id: 1 },
    address: 'P.O. Box 603, 899 Eu St.',
    isMarried: false
  },
  {
    id: 74,
    name: 'Vaughan Moody',
    age: 37,
    country: { name: 'Brazil', id: 1 },
    address: '880 Erat Rd.',
    isMarried: false
  },
  {
    id: 75,
    name: 'Randall Reeves',
    age: 44,
    country: { name: 'United Kingdom', id: 6 },
    address: '1819 Non Street',
    isMarried: false
  },
  {
    id: 76,
    name: 'Dominic Raymond',
    age: 68,
    country: { name: 'United States', id: 7 },
    address: 'Ap #689-4874 Nisi Rd.',
    isMarried: true
  },
  {
    id: 77,
    name: 'Lev Pugh',
    age: 69,
    country: { name: 'Brazil', id: 1 },
    address: 'Ap #433-6844 Auctor Avenue',
    isMarried: true
  },
  {
    id: 78,
    name: 'Desiree Hughes',
    age: 80,
    country: { name: 'France', id: 4 },
    address: '605-6645 Fermentum Avenue',
    isMarried: true
  },
  {
    id: 79,
    name: 'Idona Oneill',
    age: 23,
    country: { name: 'Russia', id: 5 },
    address: '751-8148 Aliquam Avenue',
    isMarried: true
  },
  {
    id: 80,
    name: 'Lani Mayo',
    age: 76,
    country: { name: 'United States', id: 7 },
    address: '635-2704 Tristique St.',
    isMarried: true
  },
  {
    id: 81,
    name: 'Cathleen Bonner',
    age: 40,
    country: { name: 'United States', id: 7 },
    address: '916-2910 Dolor Av.',
    isMarried: false
  },
  {
    id: 82,
    name: 'Sydney Murray',
    age: 44,
    country: { name: 'Brazil', id: 1 },
    address: '835-2330 Fringilla St.',
    isMarried: false
  },
  {
    id: 83,
    name: 'Brenna Rodriguez',
    age: 77,
    country: { name: 'China', id: 3 },
    address: '3687 Imperdiet Av.',
    isMarried: true
  },
  {
    id: 84,
    name: 'Alfreda Mcdaniel',
    age: 38,
    country: { name: 'Russia', id: 5 },
    address: '745-8221 Aliquet Rd.',
    isMarried: true
  },
  {
    id: 85,
    name: 'Zachery Atkins',
    age: 30,
    country: { name: 'United States', id: 7 },
    address: '549-2208 Auctor. Road',
    isMarried: true
  },
  {
    id: 86,
    name: 'Amelia Rich',
    age: 56,
    country: { name: 'France', id: 4 },
    address: 'P.O. Box 734, 4717 Nunc Rd.',
    isMarried: false
  },
  {
    id: 87,
    name: 'Kiayada Witt',
    age: 62,
    country: { name: 'United Kingdom', id: 6 },
    address: 'Ap #735-3421 Malesuada Avenue',
    isMarried: false
  },
  {
    id: 88,
    name: 'Lysandra Pierce',
    age: 36,
    country: { name: 'United States', id: 7 },
    address: 'Ap #146-2835 Curabitur St.',
    isMarried: true
  },
  {
    id: 89,
    name: 'Cara Rios',
    age: 58,
    country: { name: 'France', id: 4 },
    address: 'Ap #562-7811 Quam. Ave',
    isMarried: true
  },
  {
    id: 90,
    name: 'Austin Andrews',
    age: 55,
    country: { name: 'Russia', id: 5 },
    address: 'P.O. Box 274, 5505 Sociis Rd.',
    isMarried: false
  },
  {
    id: 91,
    name: 'Lillian Peterson',
    age: 39,
    country: { name: 'Canada', id: 2 },
    address: '6212 A Avenue',
    isMarried: false
  },
  {
    id: 92,
    name: 'Adria Beach',
    age: 29,
    country: { name: 'Canada', id: 2 },
    address: 'P.O. Box 183, 2717 Nunc Avenue',
    isMarried: true
  },
  {
    id: 93,
    name: 'Oleg Durham',
    age: 80,
    country: { name: 'France', id: 4 },
    address: '931-3208 Nunc Rd.',
    isMarried: false
  },
  {
    id: 94,
    name: 'Casey Reese',
    age: 60,
    country: { name: 'France', id: 4 },
    address: '383-3675 Ultrices, St.',
    isMarried: false
  },
  {
    id: 95,
    name: 'Kane Burnett',
    age: 80,
    country: { name: 'United States', id: 7 },
    address: '759-8212 Dolor. Ave',
    isMarried: false
  },
  {
    id: 96,
    name: 'Stewart Wilson',
    age: 46,
    country: { name: 'Russia', id: 5 },
    address: '718-7845 Sagittis. Av.',
    isMarried: false
  },
  {
    id: 97,
    name: 'Charity Holcomb',
    age: 31,
    country: { name: 'China', id: 3 },
    address: '641-7892 Enim. Ave',
    isMarried: false
  },
  {
    id: 98,
    name: 'Kyra Cummings',
    age: 43,
    country: { name: 'France', id: 4 },
    address: 'P.O. Box 702, 6621 Mus. Av.',
    isMarried: false
  },
  {
    id: 99,
    name: 'Stuart Wallace',
    age: 25,
    country: { name: 'Russia', id: 5 },
    address: '648-4990 Sed Rd.',
    isMarried: true
  },
  {
    id: 100,
    name: 'Carter Clarke',
    age: 59,
    country: { name: 'China', id: 3 },
    address: 'Ap #547-2921 A Street',
    isMarried: false
  }
];
